pkg/stack/unwind/unwind_table.go

```diff
@@ -130,16 +131,23 @@ func (ptb *UnwindTableBuilder) UnwindTableForPid(pid int) (UnwindTable, error) {
 		rows := buildUnwindTable(fdes)
 		level.Info(ptb.logger).Log("msg", "adding tables for mapped executable", "path", executablePath, "rows", len(rows), "low pc", fmt.Sprintf("%x", rows[0].Loc), "high pc", fmt.Sprintf("%x", rows[len(rows)-1].Loc))
 
-		// TODO(javierhonduco): Revisit this logic with PIE executables as it's only correct
-		// for non-PIE ones.
+		ASLRElegible, err := executable.IsASLRElegible(executablePath)
+		if err != nil {
+			return nil, fmt.Errorf("ASLR check failed with with: %w", err)
+		}
+
 		if strings.Contains(executablePath, mainExec) {
-			ut = append(ut, rows...)
+			if ASLRElegible {
```

Comment:
I wonder, do we need this check? Can't we add the mapping start to the locations? If it's not dynamically mapped `m.StartAddr` is probably zero.

Reply:
Yes, we need it, even if ASLR is not enabled, `StartAddr` will be non-zero, usually around `0x1000` (4k, to avoid mapping the first page, so accessing addresses `[0..firstpage_address)` will cause a SIGSGV), which is the default base address. This is already accounted for in the DWARF unwind tables, which is why we don't want to add it in this case.